uaclient/security_status.py

```diff
@@ -118,15 +120,35 @@ def get_origin_for_installed_package(
     return "third-party"
 
 
-def get_update_status(service_name: str, ua_info: Dict[str, Any]) -> str:
+@lru_cache(maxsize=None)
+def _is_candidate_version(pkg: str, version: str) -> bool:
+    """Returns True if the package version is a candidate version."""
+    candidate_version = get_pkg_candidate_version(pkg, check_esm_cache=False)
+    if candidate_version:
+        return version == candidate_version
+    return False
+
+
+def get_update_status(
+    service_name: str,
+    ua_info: Dict[str, Any],
+    version: apt_pkg.Version,
+) -> str:
     """Defines the update status for a package based on the service name.
 
     For ESM-[Infra|Apps] packages, first checks if Pro is attached. If this is
     the case, also check for availability of the service.
     """
-    if service_name in ("standard-security", "standard-updates") or (
-        ua_info["attached"] and service_name in ua_info["enabled_services"]
-    ):
+    if service_name in ("standard-security", "standard-updates"):
+        is_candidate = _is_candidate_version(
+            version.parent_pkg.name, version.ver_str
+        )
+        return (
+            UpdateStatus.AVAILABLE.value
+            if is_candidate
+            else UpdateStatus.AVAILABLE_NOT_PREFERRED.value
+        )
+    if ua_info["attached"] and service_name in ua_info["enabled_services"]:
```

Comment:
Is there a particular reason to apply the logic only to the archive, but not to possible underpreferred ESM versions?
I mean, even if those are not a planned thing, they may happen and the criteria is the same - anything available but not a candidate should be in the new state

Reply:
Updated this as per standup discussion :smile: